compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -1717,6 +1717,12 @@ fn record_layout_for_printing_outlined<'tcx>(
             return;
         }
 
+        ty::Generator(..) => {
+            debug!("print-type-size t: `{:?}` record generator", layout.ty);
+            record(DataTypeKind::Generator, false, None, vec![]);
+            return;
+        }
```

Comment:
I think most of the below code should work with generators. They have variants just like enums do.

Reply:
It'll need some massaging, since generators represent their state-specific + common types differently than ADTs do, but I'm working on it.